sdk/core/azure-core/src/main/java/com/azure/core/http/rest/ResponseExceptionConstructorCache.java

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.rest;
+
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.implementation.ReflectionUtilsApi;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Constructor;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A concurrent cache of {@link HttpResponseException} {@link MethodHandle} constructors.
+ */
+final class ResponseExceptionConstructorCache {
+    private static final Map<Class<? extends HttpResponseException>, MethodHandle> CACHE = new ConcurrentHashMap<>();
```

Comment:
should this be capped?

Reply:
Unlikely needs to be, the only path that should be calling this is the generated layer and there shouldn't be more than 100 or so error types.